sources/logdog/src/create_tarball.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::fs::File;
+use std::path::PathBuf;
+
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use snafu::ResultExt;
+use tar;
+
+// Creates a tarball with all the contents of directory `dir`.
+// Outputs as file to `outfile`.
+pub(crate) fn create_tarball(dir: &PathBuf, outfile: &PathBuf) -> crate::error::Result<()> {
```

Comment:
I misunderstood your question as "does the out file exist", which I thought was a non-issue since we just overwrite it.

No, tar is not smart enough to avoid taring its own output. I accidentally did that once. It didn't blow up the universe, but the archive had a funky file in it. Now that I get what you're saying I'll check for that case in the next push.

Reply:
Since the dir being compressed is a TempDir, this is almost an impossible case, but it doesn't hurt to check in case the code elsewhere changes someday.

Reply:
Oh, you know, you're right, I'm sorry.  It doesn't matter at all with tempdir usage.  And I can't imagine why we'd move away from tempdir, as long as we're running the commands.  (I can imagine moving to streaming the contents into a tar file instead of putting them on disk at all, but it still wouldn't matter then.)

I'm going to resolve this, I don't think there's anything to do!